library/core/src/iter/traits/collect.rs

```diff
@@ -336,7 +337,10 @@ pub trait Extend<A> {
     /// assert_eq!("abcdef", &message);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    fn extend<T: IntoIterator<Item = A>>(&mut self, iter: T);
+    fn extend<T: ~const IntoIterator<Item = A>>(&mut self, iter: T)
+    where
+        T::IntoIter: ~const Iterator<Item = A>,
+        T::IntoIter: ~const Drop;
```

Comment:
I am very scared now XD This feels like it would be a breaking change, but presumably it pans out for non-const impls?

Reply:
how is this a breaking change for non-const callers? `IntoIterator::IntoIter` already has the `Iterator` bound.

Reply:
Oh, I'm not saying it is, but just that it is scary, because my first that was that implementors of `Extend` would now have to specify more bounds, but that would have failed in liballoc on `Vec`, and since everything compiles, and since I kinda know how the logic around that works, 👍